### PR TITLE
Feat announcement

### PR DESCRIPTION
**Pull request by saenyakorn**
## Why did you create this PR
- เพื่อให้มีหน้าประกาศเป็นหลักเป็นแหล่งสักที

## What did you do
- deploy strapi CMS ที่ [cms.cugetreg.com](https://cms.cugetreg.com)
- refactor apollo for serverside / clientside caching
- สร้างหน้า announcement page /  announcement list
- รวมเข้ากับ Next 12
- สามารถเพิ่ม announcement ได้ทุกเมื่อ และสามารถ pin มันได้ด้วย 
- อัพเดท Serverside XML 
- ลดขนาด Docker Image Size ลง ประมาณ 50% ด้วย API ใหม่ของ Next 12 [Output File Tracing](https://nextjs.org/docs/advanced-features/output-file-tracing)

Screen Shot 2565-05-17 at 18 45 42

## Demo
[https://dev.cugetreg.com/announcements](https://dev.cugetreg.com/announcements)

## Checklist
- [x] Deploy a demo
- [ ] Check browsers compatibility
- [ ] Wrote coverage tests

### CODE REVIEW

**Comment by natTP** [comment]
View button doesn't work!
Just remove the button if unecessary
https://user-images.githubusercontent.com/57129145/168806522-cbd33a99-9a6f-4828-8643-9f93de8a7c4f.mp4

**Comment by saenyakorn** [comment]
> View button doesn't work! Just remove the button if unecessary https://user-images.githubusercontent.com/57129145/168806522-cbd33a99-9a6f-4828-8643-9f93de8a7c4f.mp4

missing "s" in "announcement", will fix

**Comment by saenyakorn** [comment]
Remove "seenAnnouncement" is localstorage to reproduce

**Comment by bombnp** [comment]
image
uh?

**Comment by saenyakorn** [comment]
cause 500 due to dev.cugetreg.com/api is owned by Backend, but, in this case, I want to call dev.cugetreg.com/api in Frontend (/api route in Frontend). 

Additionally, we can't change the nextjs API route name since the nextjs team [hardcode](https://stackoverflow.com/questions/60874250/nextjs-change-pages-api-to-pages-myapi) the API route name 

To solve this problem, we need to change base API path in backend

**Comment by saenyakorn** [comment]
@bombnp it works now

**Comment by saenyakorn** [comment]
> I think [title case](https://titlecaseconverter.com/) should be used here. Basically `Yesterday` (or any other word such as `Sunday` or `May 17th 2022`) should be capitalized word-by-word, but `at` `in` `on` shouldn't.
> 
> image

The solution for now is just to capitalize the first word of the relative time. See the other possible word here https://date-fns.org/v2.28.0/docs/formatRelative

File: src/common/tracker/index.ts [hunk 1]
```diff
@@ -1,12 +1,10 @@
 import TagManager from 'react-gtm-module'
```
Comment (paphonb) [inline src/common/tracker/index.ts:1]: I'm not sure if you can reduce bundle size without removing import 🤔

**Comment by saenyakorn** [comment]
Screen Shot 2565-05-27 at 01 33 27

**Review by paphonb — APPROVED** [review]

**Comment by bombnp** [comment]
is cms shared between all envs?

**Comment by bombnp** [comment]
also, is it possible to put links and make it clickable in the title?
image

**Comment by saenyakorn** [comment]
> is cms shared between all envs?

YES, CMS for all envs

**Comment by saenyakorn** [comment]
> also, is it possible to put links and make it clickable in the title? image

OK

**Pull request closed without merging**